docs/frontend-system/architecture/05-extension-overrides.md

```diff
@@ -10,38 +10,107 @@ description: Frontend extension overrides
 
 ## Introduction
 
-<!--
+An extension override is another building block in the Frontend System that allows you to programmatically override app or plugin extensions. Extension overrides are recommended in the following cases:
 
-Introduce extension overrides and how they can be shipped in separate packages but also internally within a project too.
+- Override the app skeleton layout;
+- Override a default api implementation;
+- Override a default app component;
+- Override a plugin's default page;
+- Etc.
 
-Talk about how extension overrides
+It is also important to clarify when an override isn't needed. Creating an extension override is not recommended when:
```

Comment:
I'm feeling that the real answer here is that you only need to override its implementation, and you should first make sure that you can't achieve the desired effect only through configuration instead. With the below being a few examples of how you might be able to configure an extension for your needs.

There is one interesting case to point out as well that is missing atm, and that is when the target extension has an input that allows you to configure it's behavior. You might for example have a page extension with an optional `header` input that you can use to override the default header component. In that case you would ideally create a plugin that provides the customized header extension, rather than overriding the entire page through an override.

Reply:
made it into an example and explanatory text